packages/frontend/web/client/reportError.ts

```diff
@@ -0,0 +1,26 @@
+/*
+   Report errors to Sentry with optional tags metadata
+*/
+
+import { getRaven } from '@frontend/web/client/raven';
+
+type ReportedError = Error & {
+    reported?: boolean;
+};
+
+export const reportError = (
+    err: ReportedError,
+    tags: { [key: string]: string },
+    shouldThrow: boolean = true,
+): void => {
+    getRaven().then(raven => {
+        raven.captureException(err, { tags });
+
+        if (shouldThrow) {
+            // Flag to ensure it is not reported to Sentry again via global handlers
+            const error = err;
+            error.reported = true;
+            throw error;
```

Comment:
What are we doing to ensure Sentry doesn't report the error again via global handlers? In frontend, [we override the global `onerror` handler](https://github.com/guardian/frontend/blob/ed8d4e202b1da36d6e0454d831ff506082144b26/static/src/javascripts/bootstraps/standard/main.js#L152-L157)